File: src/main/java/com/chatchatabc/jpademojava/impl/domain/service/RoleServiceImpl.java
package com.chatchatabc.jpademojava.impl.domain.service;

import com.chatchatabc.jpademojava.domain.enums.ROLE_NAMES;
import com.chatchatabc.jpademojava.domain.model.Role;
import com.chatchatabc.jpademojava.domain.repository.RoleRepository;
import com.chatchatabc.jpademojava.domain.service.RoleService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class RoleServiceImpl implements RoleService {
    @Autowired
    private RoleRepository roleRepository;

    private final Logger log = LoggerFactory.getLogger(RoleServiceImpl.class);

    /**
     * Create roles
     */
    @Override
    public void createRoles() {
        // Loop through ROLE_NAMES enum and create roles if they don't exist
        for (ROLE_NAMES role : ROLE_NAMES.values()) {
            if (roleRepository.findRoleByName(role.toString()).isEmpty()) {
                roleRepository.save(
                        Role.builder()
                                .name(role.name())
                                .build());
                log.info("Role created: {}", role.name());
            }
        }
    }
}

File: src/main/kotlin/com/chatchatabc/jpademo/impl/domain/service/RoleServiceImpl.kt
package com.chatchatabc.jpademo.impl.domain.service

import com.chatchatabc.jpademo.domain.model.ROLE_NAMES
import com.chatchatabc.jpademo.domain.model.Role
import com.chatchatabc.jpademo.domain.repository.RoleRepository
import com.chatchatabc.jpademo.domain.service.RoleService
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

@Service
class RoleServiceImpl(
    private val roleRepository: RoleRepository
) : RoleService {

    private val log = LoggerFactory.getLogger(RoleServiceImpl::class.java)

    /**
     * Create Roles on runtime
     */
    override fun createRoles() {
        // Loop through ROLE_NAMES enum and create roles if they don't exist
        ROLE_NAMES.values().forEach { roleName ->
            val role = roleRepository.findRoleByName(roleName.toString())
            if (role.isEmpty) {
                roleRepository.save(Role(name = roleName.toString()))
                log.info("Role created: $roleName")
            }
        }
    }
}